src/app/core/services/kafka.service.ts

```diff
@@ -11,30 +11,31 @@ import {
   KAFKA_COMPLETION_LOG,
   KAFKA_TIMEZONE
 } from '../../../assets/data/defaultConfig'
-import { AuthService } from '../../pages/auth/services/auth.service'
 import { StorageKeys } from '../../shared/enums/storage'
 import {
   AnswerKeyExport,
   AnswerValueExport,
   ApplicationTimeZoneValueExport,
-  CompletionLogValueExport,
+  CompletionLogValueExport
 } from '../../shared/models/answer'
 import { QuestionType } from '../../shared/models/question'
 import { Task } from '../../shared/models/task'
 import { getSeconds } from '../../shared/utilities/time'
 import { Utility } from '../../shared/utilities/util'
 import { StorageService } from './storage.service'
+import { TokenService } from './token.service'
+import { SchemaMetadata } from '../../shared/models/kafka'
 
 @Injectable()
 export class KafkaService {
   private KAFKA_CLIENT_URL: string
   private cacheSending = false
-  private schemas = {}
+  private schemas: {[key: string]: Promise<[SchemaMetadata, SchemaMetadata]>} = {}
 
   constructor(
     private util: Utility,
     private storage: StorageService,
-    private authService: AuthService
```

Comment:
Probably still need `AuthService` instead of `TokenService`, since all Keycloak-specific things are now encoded in `AuthService`.

Reply:
I agree.